analysis_driver/quality_control/calculate_relatedness.py

```diff
@@ -1,16 +1,194 @@
+            for g in gel_lines:
+                gel_outfile.write('\t'.join(g) + '\n')
+
+        with open(os.path.join(self.job_dir, self.dataset.name + '.relatedness_output.egc'), 'w') as egc_outfile:
+            egc_outfile.write(header + '\n')
+            for e in egc_lines:
+                egc_outfile.write('\t'.join(e) + '\n')
+
+
+    def get_outfile_content(self, values):
+        gel_lines = []
+        egc_lines = []
+        user_ids = self.user_sample_ids()
+        for r in values:
+            if r['sample1'] != r['sample2']:
+                comparison = {'samples': [], 'family_ids': set(), 'proband': [], 'other': []}
+
+                for sample in [r['sample1'], r['sample2']]:
+                    internal_id = user_ids[sample]
+                    comparison['samples'].append(internal_id)
+                    comparison['family_ids'].add(self.family_id(internal_id))
+                    relationship = self.relationship(internal_id)
+                    if relationship == 'Proband':
+                        comparison['proband'].append(internal_id)
+                    elif relationship != 'Proband':
+                        comparison['other'].append(internal_id)
+
+                if not len(comparison['proband'] + comparison['other']) == 2:
+                    raise PipelineError('Incorrect number of samples in this comparison')
+
+                if len(list(comparison['family_ids'])) == 1:
+                    if len(comparison['proband']) == 1:
+                        gel_line = [''.join(list(comparison['family_ids'])),
+                                comparison['proband'][0],
+                                'Proband',
+                                comparison['other'][0],
+                                self.relationship(comparison['other'])
+                                    ]
+                        gel_line.extend(r['relatedness'])
+                        gel_lines.append(gel_line)
+
+                egc_sample_pair = comparison['proband'] + comparison['other']
+                egc_line = [self.family_id(egc_sample_pair[0]),
+                            egc_sample_pair[0],
+                            self.relationship(egc_sample_pair[0]),
+                            self.family_id(egc_sample_pair[1]),
+                            egc_sample_pair[1],
+                            self.relationship(egc_sample_pair[1])
+                            ]
+
+                egc_line.extend(r['relatedness'])
+                egc_lines.append(egc_line)
+
+        return gel_lines, egc_lines
+
+
+    def get_columns(self, filename, column_headers):
+        columns_to_return = []
+        try:
+            with open(filename) as openfile:
+                csvfile = csv.DictReader(openfile)
+                for line in csvfile:
+                    columns_to_return.append([line[i] for i in column_headers])
+        except FileNotFoundError:
+            raise PipelineError('Could not find file: %s' % filename)
```

Comment:
Don't really need this, since if you don't catch the error, you'll get...
```python
>>> open('nonexistent_file.txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 2] No such file or directory: 'nonexistent_file.txt'
```
... which says exactly the same thing without needing to try/catch anything. There are a couple of other places here where this is also the case.